app/models/session.rb

```diff
@@ -11,21 +11,31 @@ def self.check_session_timeout
     purge(ttl)
   end
 
-  def self.purge(ttl)
-    ses = where("updated_at <= ?", ttl.seconds.ago.utc)
-    ses.each do|s|
+  def self.purge(ttl, batch_size = 100)
+    deleted = 0
+    loop do
```

Comment:
Yes, I was all ready to use`find_each` but didn't think it would work well with the way I'm using the entire batch to pre-load all of the users. I may be wrong about that, though. I'll re-evaluate that. Thanks!